### PR TITLE
[LM-9400] Universal feed

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**

- Added Delete Post Feature
- Added Pin Post Feature
- Added Universal Feed

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: FeedSX/customModals/ReportModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  tagId: number;
+  uuid: string;
+}
+
+// post report modal props
+interface ReportModalProps {
+  visible: boolean;
+  closeModal: () => void;
+  reportType: string;
+}
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:41]: why we have created this here and what is the need for this?

File: FeedSX/customModals/ReportModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+type Props = ReportModalProps & PostUI;
+
+const ReportModal: React.FC<Props> = props => {
+  const {visible, closeModal, reportType} = props as ReportModalProps;
+  const {postDetail} = props as PostUI;
+
+  const dispatch = useDispatch();
+  const {Id, uuid} = {...postDetail};
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:50]: rename `Id` to `id`

File: FeedSX/customModals/ReportModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  const [selectedIndex, setSelectedIndex] = useState<number>(-1);
+  const [otherReason, setOtherReason] = useState('');
+  const [selectedId, setSelectedId] = useState(-1);
+  const reportTags = useAppSelector(state => state.feed.reportTags);
+
+  // this function calls the get report tags api for reporting a post
+  const fetchReportTags = async () => {
+    let payload = {
+      type: 3, // type 3 for report tags
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:59]: use enums for this

File: FeedSX/customModals/ReportModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    };
+    let reportTagsResponse = await dispatch(
+      getReportTags(
+        GetReportTagsRequest.builder().settype(payload.type).build(),
+      ) as any,
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:64]: why as `any`?

File: FeedSX/customModals/ReportModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+      .setUuid(payload.uuid)
+      .build();
+
+    // request for reporting post with given tags
+    let reportRequest = PostReportRequest.builder()
+      .setEntityId(payload.entityId)
+      .setEntityType(payload.entityType)
+      .setTagId(payload.tagId)
+      .setUuid(payload.uuid)
+      .build();
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:99]: why two different request are created, create only one request.

File: FeedSX/customModals/ReportModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+    let postReportResponse = await dispatch(
+      postReport(payload.reason ? otherReportRequest : reportRequest) as any,
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:102]: why `as any` ?

File: FeedSX/customModals/ReportModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+    );
+    // toast message action
+    if (postReportResponse) {
+      dispatch(
+        showToastMessage({
+          isToast: true,
+          message: 'Reported Successfully',
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:109]: use string constants

File: FeedSX/customModals/ReportModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+        }) as any,
+      );
+    } else {
+      dispatch(
+        showToastMessage({
+          isToast: true,
+          message: 'Something Went Wrong',
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:116]: use string constants

File: FeedSX/customModals/ReportModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+    let postReportResponse = await dispatch(
+      postReport(payload.reason ? otherReportRequest : reportRequest) as any,
+    );
+    // toast message action
+    if (postReportResponse) {
+      dispatch(
+        showToastMessage({
+          isToast: true,
+          message: 'Reported Successfully',
+        }) as any,
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:110]: why `as any` ?

File: FeedSX/customModals/ReportModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+      );
+    } else {
+      dispatch(
+        showToastMessage({
+          isToast: true,
+          message: 'Something Went Wrong',
+        }) as any,
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:117]: why `as any` ?

File: FeedSX/customModals/ReportModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+          <Text style={styles.titleText}>Report Abuse</Text>
+          <TouchableOpacity
+            onPress={() => {
+              setSelectedId(-1);
+              setSelectedIndex(-1);
+              closeModal();
+            }}>
+            <Image
+              source={require('../../assets/images/close_icon3x.png')}
+              style={{width: 18, height: 18}}
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:151]: create a stylesheet variable for this.

File: FeedSX/customModals/ReportModal/index.tsx [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+        </View>
+
+        {/* modal content */}
+        <View style={styles.contentView}>
+          <Text style={styles.textHeading}>
+            Please specify the problem to continue
+          </Text>
+          <Text style={styles.text}>
+            You would be able to report this {reportType} after selecting a
+            problem
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:163]: add string constants

File: FeedSX/customModals/ReportModal/index.tsx [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+
+        {/* text input view for other reason text*/}
+        {selectedIndex == 5 ? (
+          <View style={styles.otherSection}>
+            <TextInput
+              onChangeText={e => {
+                setOtherReason(e);
+              }}
+              style={styles.otherTextInput}
+              placeholder="Enter the reason for Reporting this post"
```
Comment (ishaan1607) [inline FeedSX/customModals/ReportModal/index.tsx:216]: use string constants

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 14]
```diff
@@ -1,27 +1,38 @@
...
+  const [feedPageNumber, setFeedPageNumber] = useState(1);
   const [communityId, setCommunityId] = useState('');
   const [accessToken, setAccessToken] = useState('');
 
   // this function calls initiate user API and sets the access token and community id
   async function getInitialData() {
     //this line of code is for the sample app only, pass your userUniqueID instead of this.
     // const UUID = await AsyncStorage.getItem('userUniqueID');
-    const UUID = '10003';
+        const UUID = '10003';
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:35]: add a todo message that we are using a dummy value.

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 15]
```diff
@@ -57,9 +69,42 @@ const UniversalFeed = () => {
...
+    }
+    return postLikeResponse;
+  }
+
+  // this functions hanldes the post save functionality
+  async function savePostHandler(id: string) {
+    let payload = {
+      postId: id,
+    };
+    dispatch(savePostStateHandler(payload.postId) as any);
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:94]: why as any?

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 16]
```diff
@@ -57,9 +69,42 @@ const UniversalFeed = () => {
...
+    // calling the save post api
+    let savePostResponse = await dispatch(
+      savePost(
+        SavePostRequest.builder().setpostId(payload.postId).build(),
+      ) as any,
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:99]: why as any?

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 17]
```diff
@@ -57,9 +69,42 @@ const UniversalFeed = () => {
...
+  // this functions hanldes the post save functionality
+  async function savePostHandler(id: string) {
+    let payload = {
+      postId: id,
+    };
+    dispatch(savePostStateHandler(payload.postId) as any);
+    // calling the save post api
+    let savePostResponse = await dispatch(
+      savePost(
+        SavePostRequest.builder().setpostId(payload.postId).build(),
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:98]: it should `setPostId()`

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 18]
```diff
@@ -57,9 +69,42 @@ const UniversalFeed = () => {
...
+  // this functions hanldes the post like functionality
+  async function postLikeHandler(id: string) {
+    let payload = {
+      postId: id,
+    };
+    dispatch(likePostStateHandler(payload.postId) as any);
+    // calling like post api
+    let postLikeResponse = await dispatch(
+      likePost(
+        LikePostRequest.builder().setpostId(payload.postId).build(),
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:81]: it should `setPostId()`

File: FeedSX/store/actions/feed.ts [hunk 19]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
         showLoader: true,
       },
     });
   } catch (error) {
     Alert.alert(`${error}`);
   }
 };
+
+// like post API action
+export const likePost = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:102]: why any?

File: FeedSX/store/actions/feed.ts [hunk 20]
```diff
@@ -10,16 +10,42 @@ import {
...
+  PIN_POST_STATE,
+  DELETE_POST_STATE,
 } from '../types/types';
-import { lmFeedClient } from '../../..';
-import { InitiateUserRequest } from 'likeminds-sdk';
+import {lmFeedClient} from '../../..';
+import {InitiateUserRequest} from 'likeminds-sdk';
 
 // initiateUser API action
 export const initiateUser = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:41]: why any?

File: FeedSX/store/actions/feed.ts [hunk 21]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+      },
+    });
+  } catch (error) {
+    Alert.alert(`${error}`);
+  }
+};
+
+// like post state managing action
+export const likePostStateHandler =
+  (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:120]: why any?

File: FeedSX/store/actions/feed.ts [hunk 22]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+        body: payload,
+      });
+      return;
+    } catch (error) {
+      Alert.alert(`${error}`);
+    }
+  };
+
+// save post API action
+export const savePost = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:133]: why any?

File: FeedSX/store/actions/feed.ts [hunk 23]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+        showLoader: true,
+      },
+    });
+  } catch (error) {
+    Alert.alert(`${error}`);
+  }
+};
+
+// save post state managing action
+export const savePostStateHandler =
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:150]: why any?

File: FeedSX/store/actions/feed.ts [hunk 24]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+        body: payload,
+      });
+      return;
+    } catch (error) {
+      Alert.alert(`${error}`);
+    }
+  };
+
+// get report tags API action
+export const getReportTags = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:164]: why any?

File: FeedSX/store/actions/feed.ts [hunk 25]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+        showLoader: true,
+      },
+    });
+  } catch (error) {
+    Alert.alert(`${error}`);
+  }
+};
+
+// report post API action
+export const postReport = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:181]: why any?

File: FeedSX/store/actions/feed.ts [hunk 26]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+        showLoader: true,
+      },
+    });
+  } catch (error) {
+    Alert.alert(`${error}`);
+  }
+};
+
+// delete post API action
+export const deletePost = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:198]: why any?

File: FeedSX/store/actions/feed.ts [hunk 27]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+      },
+    });
+  } catch (error) {
+    Alert.alert(`${error}`);
+  }
+};
+
+// delete post state managing action
+export const deletePostStateHandler =
+  (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:216]: why any?

File: FeedSX/store/actions/feed.ts [hunk 28]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+        body: payload,
+      });
+      return;
+    } catch (error) {
+      Alert.alert(`${error}`);
+    }
+  };
+
+// pin post API action
+export const pinPost = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:229]: why any?

File: FeedSX/store/actions/feed.ts [hunk 29]
```diff
@@ -59,11 +85,173 @@ export const getFeed = (payload?: any) => async (dispatch: Dispatch) => {
...
+      },
+    });
+  } catch (error) {
+    Alert.alert(`${error}`);
+  }
+};
+
+// pin post state managing action
+export const pinPostStateHandler =
+  (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/feed.ts:247]: why any?

File: FeedSX/store/actions/toast.ts [hunk 30]
```diff
@@ -0,0 +1,15 @@
+import {Dispatch} from '@reduxjs/toolkit';
+import {Alert} from 'react-native';
+import { SHOW_TOAST } from '../types/loader';
+
+// show toast message action
+export const showToastMessage = (payload?: any) => async (dispatch: Dispatch) => {
```
Comment (ishaan1607) [inline FeedSX/store/actions/toast.ts:6]: why any?

File: FeedSX/store/reducers/feedReducer.ts [hunk 31]
```diff
@@ -1,13 +1,27 @@
...
+  PIN_POST_SUCCESS,
+  PIN_POST_STATE,
+  DELETE_POST_STATE,
 } from '../types/types';
 
 const initialState = {
+  // todo: remove any and define its type
   community: {},
-  user: {},
+  users: {} as any,
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:20]: why any?

File: FeedSX/store/reducers/feedReducer.ts [hunk 32]
```diff
@@ -1,13 +1,27 @@
...
   memberRights: [],
-  feed: {}
+  member: {},
+  feed: [] as any,
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:23]: why any?

File: FeedSX/store/reducers/feedReducer.ts [hunk 33]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+      return {...state, feed: feedData, users: usersData};
+    }
+    case LIKE_POST_SUCCESS: {
+      return {...state};
+    }
+    case LIKE_POST_STATE: {
+      let updatedFeed = state.feed;
+      // this gets the index of post that is liked
+      const likedPostIndex = updatedFeed.findIndex(
+        (item: any) => item?.Id === action.body,
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:54]: why any?

File: FeedSX/store/reducers/feedReducer.ts [hunk 34]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+      return {...state, feed: updatedFeed};
+    }
+    case SAVE_POST_SUCCESS: {
+      return {...state};
+    }
+    case SAVE_POST_STATE: {
+      let updatedFeed = state.feed;
+      // this gets the index of post that is saved
+      const savedPostIndex = updatedFeed.findIndex(
+        (item: any) => item?.Id === action.body,
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:77]: why any?

File: FeedSX/store/reducers/feedReducer.ts [hunk 35]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+      return {...state, feed: updatedFeed};
+    }
+    case PIN_POST_SUCCESS: {
+      return {...state};
+    }
+    case PIN_POST_STATE: {
+      let updatedFeed = state.feed;
+      // this gets the index of post that is pinned
+      const pinnedPostIndex = updatedFeed.findIndex(
+        (item: any) => item?.Id === action.body,
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:92]: why any?

File: FeedSX/store/reducers/feedReducer.ts [hunk 36]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+      );
+      // this updates the isPinned value
+      updatedFeed[pinnedPostIndex]['isPinned'] =
+        !updatedFeed[pinnedPostIndex]['isPinned'];
+      // this gets the index of pin/unpin from menu item
+      let menuItemIndex = updatedFeed[pinnedPostIndex]['menuItems'].findIndex(
+        (item: any) => item.id === 2 || item.id === 3,
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:99]: why any?

Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:99]: use enums for id checks

File: FeedSX/store/reducers/feedReducer.ts [hunk 37]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+      );
+      if (updatedFeed[pinnedPostIndex]['isPinned']) {
+        //  this updates the menuItem title to unpin
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].id = 3;
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:103]: use enums for id checks

File: FeedSX/store/reducers/feedReducer.ts [hunk 38]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].title =
+          'Unpin this Post';
+      } else {
+        //  this updates the menuItem title to pin
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].id = 2;
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:108]: use enums for id checks

File: FeedSX/store/reducers/feedReducer.ts [hunk 39]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+        !updatedFeed[pinnedPostIndex]['isPinned'];
+      // this gets the index of pin/unpin from menu item
+      let menuItemIndex = updatedFeed[pinnedPostIndex]['menuItems'].findIndex(
+        (item: any) => item.id === 2 || item.id === 3,
+      );
+      if (updatedFeed[pinnedPostIndex]['isPinned']) {
+        //  this updates the menuItem title to unpin
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].id = 3;
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].title =
+          'Unpin this Post';
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:105]: use string constants

File: FeedSX/store/reducers/feedReducer.ts [hunk 40]
```diff
@@ -18,11 +32,105 @@ export function feedReducer(state = initialState, action: any) {
...
+      } else {
+        //  this updates the menuItem title to pin
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].id = 2;
+        updatedFeed[pinnedPostIndex]['menuItems'][menuItemIndex].title =
+          'Pin this Post';
```
Comment (ishaan1607) [inline FeedSX/store/reducers/feedReducer.ts:110]: use string constants

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: FeedSX/components/LMPost/VideoPost/index.tsx [hunk 41]
```diff
@@ -4,14 +4,18 @@ import {PostUI} from '../../../Models/PostModel';
 import styles from './styles';
 import {VIDEO_ATTACHMENT_TYPE} from '../../../constants/Strings';
 
-const VideoPost = ({postAttachments}: PostUI) => {
+const VideoPost = ({postDetail}: PostUI) => {
```
[the remark this answers is not in the document]
Comment (prabha-likeminds) [reply FeedSX/components/LMPost/VideoPost/index.tsx:7]: this is a component name so this the standard format of naming it (same is followed for all components)

File: FeedSX/components/LMPost/index.tsx [hunk 42]
```diff
@@ -26,47 +26,39 @@ const LMPost = ({
...
-  postMenuItems,
   carauselActiveItemColor,
   carauselInActiveItemColor,
   carouselPaginationStyle,
+  postDetail,
+  postUserDetail,
 }: PostUI) => {
-  const [liked, setLiked] = useState(likedState);
-  const [saved, setSaved] = useState(savedState);
+  const {Id, attachments, isLiked, isSaved, text} = {...postDetail};
```
[the remark this answers is not in the document]
Comment (prabha-likeminds) [reply FeedSX/components/LMPost/index.tsx:59]: we are getting this key from backend like this only.

File: FeedSX/customModals/ReportModal/index.tsx (hunk 1 above)
Comment (prabha-likeminds) [reply FeedSX/customModals/ReportModal/index.tsx:41]: We need data from two interfaces over here so this interface defines the props that this modal expects for handling its visibilty and the content in the modal is shown from the PostUI interface, which the 2nd interface.

File: FeedSX/customModals/DeleteModal/index.tsx [hunk 43]
```diff
@@ -0,0 +1,172 @@
...
+}
+type Props = DeleteModalProps & PostUI;
+
+const DeleteModal: React.FC<Props> = props => {
+  const {visible, displayModal, deleteType} = props as DeleteModalProps;
+  const {postDetail, modalBackdropColor, postUserDetail} = props as PostUI;
+
+  const dispatch = useDispatch();
+  const loggedInUser = useAppSelector(state => state.feed.member);
+  const {Id, userId} = {...postDetail};
```
[the remark this answers is not in the document]
Comment (prabha-likeminds) [reply FeedSX/customModals/DeleteModal/index.tsx:35]: we are getting this key from backend like this only.

File: FeedSX/customModals/ReportModal/index.tsx (hunk 2 above)
Comment (prabha-likeminds) [reply FeedSX/customModals/ReportModal/index.tsx:50]: we are getting this key from backend like this only.

File: FeedSX/screens/UniversalFeed/index.tsx (hunk 18 above)
Comment (prabha-likeminds) [reply FeedSX/screens/UniversalFeed/index.tsx:81]: its automatically suggesting setpostId only

File: FeedSX/store/reducers/feedReducer.ts (hunk 39 above)
Comment (prabha-likeminds) [reply FeedSX/store/reducers/feedReducer.ts:105]: resolved

File: FeedSX/store/reducers/feedReducer.ts (hunk 38 above)
Comment (prabha-likeminds) [reply FeedSX/store/reducers/feedReducer.ts:108]: resolved

File: FeedSX/store/reducers/feedReducer.ts (hunk 37 above)
Comment (prabha-likeminds) [reply FeedSX/store/reducers/feedReducer.ts:103]: resolved

File: FeedSX/store/reducers/feedReducer.ts (hunk 36 above)
Comment (prabha-likeminds) [reply FeedSX/store/reducers/feedReducer.ts:99]: resolved

File: FeedSX/customModals/ReportModal/index.tsx (hunk 3 above)
Comment (prabha-likeminds) [reply FeedSX/customModals/ReportModal/index.tsx:59]: resolved

File: FeedSX/components/LMLoader/index.tsx [hunk 44]
```diff
@@ -0,0 +1,8 @@
+import {ActivityIndicator, View} from 'react-native';
+import React from 'react';
+
+const LMLoader = () => {
+  return <ActivityIndicator size="large" color={'#5046E5'} />;
```
[the remark this answers is not in the document]
Comment (prabha-likeminds) [reply FeedSX/components/LMLoader/index.tsx:5]: resolved

File: FeedSX/customModals/ReportModal/index.tsx (hunk 5 above)
Comment (prabha-likeminds) [reply FeedSX/customModals/ReportModal/index.tsx:99]: resolved

File: FeedSX/customModals/ReportModal/index.tsx (hunk 12 above)
Comment (prabha-likeminds) [reply FeedSX/customModals/ReportModal/index.tsx:163]: resolved

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**